helper/testhelpers/testhelpers.go

```diff
@@ -650,37 +650,44 @@ func WaitForActiveNodeAndPerfStandbys(t testing.T, cluster *vault.TestCluster) {
 	var standbys, actives int64
 	var wg sync.WaitGroup
 	deadline := time.Now().Add(30 * time.Second)
-	for _, c := range cluster.Cores {
+	for i, c := range cluster.Cores {
 		wg.Add(1)
-		go func(client *api.Client) {
+		go func(coreIdx int, client *api.Client) {
 			defer wg.Done()
 			val := 1
 			for time.Now().Before(deadline) {
-				_, err = cluster.Cores[0].Client.Logical().Write(path, map[string]interface{}{
-					"bar": val,
-				})
-				if err != nil {
-					t.Fatal("unable to write KV", "path", path)
+				if coreIdx == 0 {
+					_, err = cluster.Cores[0].Client.Logical().Write(path, map[string]interface{}{
+						"bar": val,
+					})
+					if err != nil {
+						t.Fatal("unable to write KV", "path", path)
+					}
 				}
 				val++
 				time.Sleep(250 * time.Millisecond)
-				leader, err := client.Sys().Leader()
-				if err != nil {
-					if strings.Contains(err.Error(), "Vault is sealed") {
-						continue
+
+				if coreIdx > 0 || atomic.LoadInt64(&actives) == 0 {
```

Comment:
I think this conditional is now eliminating a case that we were previously covering. If a standby (non-0th core) becomes the leader first (and that goroutine happens to run first), then we increment the active counter. Once the goroutine for core0 runs through, we don't increment the active count since it's non-zero, so the check on L686 (`if actives == 1 ...`) won't trip.

Reply:
You're right that we're assuming that the 0th core is always the active core.  In the absence of stepdowns or other shenanigans I believe we can assume that, and there's plenty of test code that does.  Note that in NewTestCluster it does a TestWaitActive on core0 before it unseals the other cores.  I will add some godoc to clarify that this helper relies on that assumption.

Reply:
I think it's fine to assume that core0 is the active core. My concern was on this helper no longer failing in the case where we assume core0 is the active but a standby actually becomes active first due to a bug or ordering issue (and active not getting incremented to 2).

Reply:
NewTestCluster will fail if core0 doesn't become active, so for any regular usage this helper does its job of keeping tests from proceeding before all cores are ready (active or standbys.)  I decided against trying to handle the use case you're describing because I've run out of time for working on tests and this solves the problem it was created to solve, i.e. making our existing tests more reliable.  We can always improve it further later if so desired.

Reply:
Sounds good, adding godoc around this should suffice then.